packages/loopring_v3/contracts/impl/libexchange/ExchangeAdmins.sol

```diff
@@ -205,7 +205,22 @@ library ExchangeAdmins
         returns (uint)
     {
         require(!S.isInWithdrawalMode(), "INVALID_MODE");
-        return durationMinutes.mul(S.loopring.downtimePriceLRCPerMinute());
+
+        // Initially, the penalty is the percentage of DEX's downtime.
+        uint penalty = S.totalTimeInMaintenanceSeconds.mul(100) /
```

Comment:
I think this should be e.g. `.mul(10000)` because a minimum downtime of 1% is already very bad downtime.
I also think we should calculate with the extra downtime minutes already included (otherwise it's better to buy a lot of downtime when the penalty is still low), so something like this:
```
uint penalty = S.totalTimeInMaintenanceSeconds.add(durationMinutes.mul(60)).mul(10000) / 
     (now - S.exchangeCreationTimestamp);
```

Reply:
I knew this, and I want to give DEX owner the chance to buy more downtime than initially needed with a discount.  This will end up locking up extra LRC.

Reply:
For business people, they just need to remember under 1%, the downtime price is the same. We don’t need to fine tune the price based in smaller granularity, it will probably confuse people. 1% means in one year, the DEX can be taken off chain for 3 days and a half.

Reply:
I'm okay with this is if the price for downtime is very high to begin with. Though I still think it's a good idea to increase the price the more downtime is purchased even in a single purchase, security of user funds should come first here.

This is a risky feature. User funds are safe, but the DEX owner can lock them up at any time for an unspecified time. If the downtime cost starts at $1000/hour, and the DEX buys the downtime in a single purchase when the price is low, than it costs _only_ $8.5M to lock up everybody's money for 1 year. Binance can lose $40M in a hack without any real problem... I can't really think of a good reason why a DEX would do this, but it can be done so it is a risk for users (and the exchange loses some of the non-custodialness because of this).

Reply:
If the downtime calculation is more complicated than this one, I prefer moving it to another contract (`DowntimePriceCalculator` or a better name?) for upgradability.
It will have a method:

```

getDowntimePrice(
    uint totalTimeInMaintenanceSeconds,
    uint totalDEXLifetimeSeconds,
    uint availableDowntimeMinutes,
    uint amountOfLRCStakedbyOwner,
    uint durationToPurchaseMinutes
    )
    external
    view
    returns (uint priceAsLRCPerMinute);
```